### PR TITLE
Change EL7 to EL8 as main testing platform

### PR DESCRIPTION
**Pull request by tatb-vmware**
PXF 7 plans to deprecate CentOS 7 as a supported platform to be in line with Greenplum 7 compatibility.

This PR does the following:
- remove CentOS 7 as the main testing platform from the pipelines and replaces it with Rocky 8
- change upgrade test to upgrade from PXF 6 -> PXF 7

Other changes include:
- bootstrap the `cluster migrate` command to move the config files that were created in `PXF_HOME` in `setup-pxf-6-latest` task to `PXF_BASE`. This only applies to the upgrade task.
- remove PXF 5 because it is not longer used for the upgrade test for PXF 5 -> PXF 6
- remove conditional checks for PXF 5 since it will be removed
- use `${DEFAULT_OS_USER}` instead of hardcoding OS name
- install `nfs-utils` in `install_nfs.bash` because the Rocky 8 image does not come with tools to set up NFS by default (CentOS 7 did come with said tools)
- standardize all the spacing in `install_pxf.bash` (before it was mixed spaces and tabs, now it is all tabs)

### CODE REVIEW

File: concourse/pipelines/templates/jobs/test-multinode-tpl.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@
       trigger: true
 {% if x.with_upgrade == 'true' %}
     - get: pxf_artifact
-      resource: pxf5-gp6-el7-released   # for upgrade test
+      resource: pxf6-gp6-el8-released   # for upgrade test, pxf 5-gp6-el8 does not exist and we do not support
```
Comment (axue-broadcom) [inline concourse/pipelines/templates/jobs/test-multinode-tpl.yml:38]:
```suggestion
      resource: pxf6-gp6-el8-released   # for upgrade test
```

File: concourse/scripts/install_nfs.bash [hunk 2]
```diff
@@ -21,6 +21,8 @@ function create_nfs_installer_scripts() {
 
 set -euxo pipefail
 
+sudo dnf install -y nfs-utils
```
Comment (axue-broadcom) [inline concourse/scripts/install_nfs.bash:24]: add a comment on why this is needed now

File: concourse/scripts/install_nfs.bash [hunk 3]
```diff
@@ -48,7 +50,9 @@ EOFF
 function run_nfs_installation() {
 
   # install and configure the NFS server on coordinator
-  ssh "centos@${COORDINATOR_HOSTNAME}" "
+  ssh "rocky@${COORDINATOR_HOSTNAME}" "
```
Comment (axue-broadcom) [inline concourse/scripts/install_nfs.bash:53]: so we don't have to go changing this in the future again if we change the OS. can we do something similar to `DEFAULT_OS_USER`?

File: concourse/pipelines/templates/jobs/test-multinode-tpl.yml [hunk 4]
```diff
@@ -7,7 +7,7 @@
  # x.target     - the target external system type
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/jobs/test-multinode-tpl.yml:7]: also run the `test-pxf-cli` on el8

File: concourse/pipelines/templates/resources/pxf6-gp6-el8-artifact-tpl.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{# template for a pxf6 gp6 el8 build artifact resource definition.
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/resources/pxf6-gp6-el8-artifact-tpl.yml:1]: in the old PXF resource file we capitalized these words
```suggestion
{# template for a PXF6 GP6 EL8 build artifact resource definition.
```

File: concourse/pipelines/templates/resources/pxf6-gp6-el8-artifact-tpl.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  #
+  # does not expect any variables to be set.
+  #}
+
+  # pxf5-gp7-el8 does not exist and we do not support it
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/resources/pxf6-gp6-el8-artifact-tpl.yml:6]: is this comment still necessary?

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 7]
```diff
@@ -48,8 +48,8 @@
 #}
 
 {% set build_test_pxf_combinations = [
-    {'gp_ver': '6', 'build_os': 'centos', 'test_os': 'centos', 'os_ver': '7',     'test_fdw': false, 'test_file': true , 'test_cli': false, 'test_multi': true , 'test_features': supported_features + ['upgrade-extension'], 'generate_release_tarball': true},
-    {'gp_ver': '6', 'build_os': 'rocky' , 'test_os': 'rocky' , 'os_ver': '8',     'test_fdw': true , 'test_file': false, 'test_cli': false, 'test_multi': false, 'test_features': [], 'generate_release_tarball': false},
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/build_pipeline-tpl.yml:52]: can you also change the line 13 comment in the file `pxf6_minor-rpm-artifact-tpl.yml` to be rocky8?

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 8]
```diff
@@ -249,13 +249,13 @@ jobs:
         {% do x.update({'use_impers': 'true'}) %}
         {% do x.update({'target': default_distro}) %}
         {% do x.update({'distro': default_distro}) %}
-        {# include upgrade task when running on centos7 only #}
-        {% set with_upgrade = 'true' if x.test_platform == 'centos7' else 'false' %}
+        {# include upgrade task when running on Rocky8 only #}
+        {% set with_upgrade = 'true' if x.test_platform == 'rocky8' and x.gp_ver == '6' else 'false' %}
```
Comment (yimingli-vmware) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:253]: on line 258 the `x.gp_ver` is in front of `x.test_platform`, let's keep the order consistent

File: concourse/scripts/upgrade_pxf.bash [hunk 9]
```diff
@@ -66,3 +69,5 @@ function _main() {
 }
 
 _main
+
+echo "Placeholder for PXF 6 to PXF 7"
```
Comment (yimingli-vmware) [inline concourse/scripts/upgrade_pxf.bash:73]: Why we need this echo here?

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml [hunk 10]
```diff
@@ -34,8 +34,8 @@
 #}
 
 {% set build_test_pxf_combinations = [
-    {'gp_ver': '6', 'build_os': 'centos', 'test_os': 'centos', 'os_ver': '7',     'test_fdw': false, 'test_file': true , 'test_cli': false, 'test_multi': true , 'test_features': common_features},
-    {'gp_ver': '6', 'build_os': 'rocky' , 'test_os': 'rocky' , 'os_ver': '8',     'test_fdw': true , 'test_file': false, 'test_cli': false, 'test_multi': false, 'test_features': []},
+    {'gp_ver': '6', 'build_os': 'centos', 'test_os': 'centos', 'os_ver': '7',     'test_fdw': false, 'test_file': false , 'test_cli': false, 'test_multi': true , 'test_features': []},
```
Comment (denalex) [inline concourse/pipelines/templates/dev_build_pipeline-tpl.yml:37]:
```suggestion
    {'gp_ver': '6', 'build_os': 'centos', 'test_os': 'centos', 'os_ver': '7',     'test_fdw': false, 'test_file': false , 'test_cli': false, 'test_multi': false , 'test_features': []},
```

File: concourse/pipelines/templates/jobs/test-multinode-tpl.yml [hunk 11]
```diff
@@ -7,7 +7,7 @@
...
  # x.distro     - the distro to use for a singlecluster image
  # x.passed     - an array of upstream job names that the artifacts need to pass before this job (optional)
- # x.with_upgrade - 'true' is the job should include an upgrade task from PXF5 to PXF6
+ # x.with_upgrade - 'true' is the job should include an upgrade task from PXF6 to PXF latest
```
Comment (denalex) [inline concourse/pipelines/templates/jobs/test-multinode-tpl.yml:10]:
```suggestion
 # x.with_upgrade - 'true' is the job should include an upgrade task from PXF6 to PXF7
```

File: concourse/pipelines/templates/jobs/test-multinode-tpl.yml [hunk 12]
```diff
@@ -170,8 +170,7 @@
       INSTALL_GPHDFS: false
       KERBEROS: ((kerberos-enabled))
       PXF_JVM_OPTS: ((pxf-jvm-opts))
-      PXF_VERSION: 5
-  - task: upgrade-to-pxf-6
+  - task: upgrade-to-pxf-latest
```
Comment (denalex) [inline concourse/pipelines/templates/jobs/test-multinode-tpl.yml:173]:
```suggestion
  - task: upgrade-to-pxf-7
```

File: concourse/scripts/install_pxf.bash [hunk 13]
```diff
@@ -273,7 +273,7 @@ function run_pxf_installer_scripts() {
 	"
 
 	# Create a database for PXF extension upgrade testing
-	if [[ ${PXF_VERSION} == 5 ]]; then
+	if [[ ${PXF_VERSION} == 6 ]]; then
```
Comment (denalex) [inline concourse/scripts/install_pxf.bash:276]: there is a similar check on line 194. We also need to get rid of `PXF_CONF` stuff for PXF5, not sure if that is out of scope

Also, do we need to explicitly pass PXF_VERSION as a job parameter ?

File: concourse/scripts/upgrade_pxf.bash [hunk 14]
```diff
@@ -8,35 +8,38 @@ GPHOME=/usr/local/greenplum-db-devel
...
 echoGreen() { echo $'\e[0;32m'"$1"$'\e[0m'; }
 
 function upgrade_pxf() {
-	echoGreen "Stopping PXF 5"
-	ssh "${COORDINATOR_HOSTNAME}" "${PXF5_HOME}/bin/pxf version && ${PXF5_HOME}/bin/pxf cluster stop"
+	echoGreen "Stopping PXF 6"
+	ssh "${COORDINATOR_HOSTNAME}" "${PXF6_HOME}/bin/pxf version && ${PXF6_HOME}/bin/pxf cluster stop"
 
-	echoGreen "Installing PXF 6"
+	echoGreen "Installing PXF latest"
```
Comment (denalex) [inline concourse/scripts/upgrade_pxf.bash:30]:
```suggestion
	echoGreen "Installing PXF 7"
```
here and everywhere else -- let's use PXF 7 explicitly (and not PXF latest) as this is what we will be upgrading to, not PXF 6 latest

File: concourse/scripts/upgrade_pxf.bash [hunk 15]
```diff
@@ -50,10 +53,10 @@ function upgrade_pxf() {
 		"
 	fi
 
-	echoGreen "Perform PXF migrate from PXF_CONF=~gpadmin/pxf to PXF_BASE_DIR=${PXF_BASE_DIR}"
-	ssh "${COORDINATOR_HOSTNAME}" "PXF_BASE=${PXF_BASE_DIR} PXF_CONF=~gpadmin/pxf ${PXF_HOME}/bin/pxf cluster migrate"
+	echoGreen "Perform PXF migrate from PXF_HOME to PXF_BASE_DIR=${PXF_BASE_DIR}"
+	ssh "${COORDINATOR_HOSTNAME}" "PXF_BASE=${PXF_BASE_DIR} PXF_CONF=${PXF_HOME} ${PXF_HOME}/bin/pxf cluster migrate"
```
Comment (denalex) [inline concourse/scripts/upgrade_pxf.bash:57]:
```suggestion
	ssh "${COORDINATOR_HOSTNAME}" "PXF_BASE=${PXF_BASE_DIR} ${PXF_HOME}/bin/pxf cluster migrate"
```
this assumes we will have PXF6 --> PXF7 migrate command, which I think we should have. In any case PXF_CONF does not make sense in a context where there's no PXF5.

File: version [hunk 16]
```diff
@@ -1 +1 @@
-6.10.1-SNAPSHOT
+6.99.99-SNAPSHOT
```
Comment (denalex) [inline version:1]: what is this ? We can do `7.0.0-SNAPSHOT` instead

File: concourse/scripts/upgrade_pxf.bash (hunk 9 above)
Comment (denalex) [inline concourse/scripts/upgrade_pxf.bash:73]: why ? The whole migration script above should be for PXF6 -> PXF7

Comment (tatb-vmware) [reply concourse/scripts/upgrade_pxf.bash:73]: It is from a prior iteration of this script, it is not needed now.

File: version (hunk 16 above)
Comment (tatb-vmware) [reply version:1]: The change was to get the upgrade script to work. We can do `7.0.0-SNAPSHOT` and it will work as well.

File: concourse/scripts/upgrade_pxf.bash (hunk 15 above)
Comment (tatb-vmware) [reply concourse/scripts/upgrade_pxf.bash:57]: We are bootstrapping the `cluster migrate` command to move the config files that were created in `PXF_HOME` in the `setup-pxf-6-latest` task to `PXF_BASE`. The alternate here is to set up `PXF_BASE` in the `setup-pxf-6 task-latest` and remove this line as we do not want to rewrite `cluster migrate` as it is out of scope for this PR.

File: concourse/pipelines/templates/build_pipeline-tpl.yml [hunk 17]
```diff
@@ -48,8 +48,8 @@
...
+    {'gp_ver': '6', 'build_os': 'centos', 'test_os': 'centos', 'os_ver': '7',     'test_fdw': false, 'test_file': false , 'test_cli': false, 'test_multi': true , 'test_features': [], 'generate_release_tarball': true},
```
[the remark this answers is not in the document]
Comment (tatb-vmware) [reply concourse/pipelines/templates/build_pipeline-tpl.yml:51]: This PR is to switch our main testing platform to `rocky8` so we want to keep our current testing features intact. If we removed testing multi-cluster on `centos7`, we would break the 'MULTINODE_EL7' flag in our Makefile. We will have a separate PR to remove `centos7` for PXF 7.

File: concourse/pipelines/templates/dev_build_pipeline-tpl.yml (hunk 10 above)
Comment (tatb-vmware) [reply concourse/pipelines/templates/dev_build_pipeline-tpl.yml:37]: Above comment

File: concourse/scripts/install_pxf.bash (hunk 13 above)
Comment (axue-broadcom) [reply concourse/scripts/install_pxf.bash:276]: Correct. However, eventually the default version of `PXF_VERSION` will be 7 and the setup task will require the PXF_VERSION to be set explicitly.

File: concourse/scripts/install_nfs.bash [hunk 18]
```diff
@@ -50,8 +54,9 @@ EOFF
 function run_nfs_installation() {
 
   # install and configure the NFS server on coordinator
-  ssh "rocky@${COORDINATOR_HOSTNAME}" "
+  ssh "${DEFAULT_OS_USER}@${COORDINATOR_HOSTNAME}" "
```
Comment (axue-broadcom) [inline concourse/scripts/install_nfs.bash:57]: i still see places below on line 81 and 82 where we hardcode rocky. can we also pass `${DEFAULT_OS_USER}` there as well?

File: concourse/scripts/upgrade_pxf.bash (hunk 15 above)
Comment (yimingli-vmware) [inline concourse/scripts/upgrade_pxf.bash:57]:
```suggestion
	ssh "${COORDINATOR_HOSTNAME}" "PXF_BASE=${PXF_BASE_DIR} PXF_CONF=${PXF6_HOME} ${PXF_HOME}/bin/pxf cluster migrate"
```
Should we set the `PXF_CONF` to be the path of the old PXF?

File: concourse/scripts/upgrade_pxf.bash [hunk 19]
```diff
@@ -50,10 +53,10 @@ function upgrade_pxf() {
 		"
 	fi
 
-	echoGreen "Perform PXF migrate from PXF_CONF=~gpadmin/pxf to PXF_BASE_DIR=${PXF_BASE_DIR}"
-	ssh "${COORDINATOR_HOSTNAME}" "PXF_BASE=${PXF_BASE_DIR} PXF_CONF=~gpadmin/pxf ${PXF_HOME}/bin/pxf cluster migrate"
+	echoGreen "Perform PXF migrate from PXF_HOME to PXF_BASE_DIR=${PXF_BASE_DIR}"
```
Comment (yimingli-vmware) [inline concourse/scripts/upgrade_pxf.bash:56]:
```suggestion
	echoGreen "Perform PXF migrate from PXF_HOME=${PXF6_HOME} to PXF_BASE_DIR=${PXF_BASE_DIR}"
```

Comment (tatb-vmware) [reply concourse/scripts/upgrade_pxf.bash:56]: After discussion with @yimingli-vmware, we have decided to keep what we currently have and change these once we have a way for users to migrate from PXF 6 to PXF 7. Then, we will revisit the upgrade pxf task.

File: concourse/scripts/upgrade_pxf.bash (hunk 15 above)
Comment (tatb-vmware) [reply concourse/scripts/upgrade_pxf.bash:57]: After discussion with @yimingli-vmware, we have decided to keep what we currently have and change these once we have a way for users to migrate from PXF 6 to PXF 7. Then, we will revisit the upgrade pxf task.

**Review by djianwen-vmware — APPROVED** [review]

**Review by yimingli-vmware — APPROVED** [review]

File: concourse/pipelines/templates/jobs/test-multinode-tpl.yml [hunk 20]
```diff
@@ -35,7 +35,7 @@
       trigger: true
 {% if x.with_upgrade == 'true' %}
     - get: pxf_artifact
-      resource: pxf5-gp6-el7-released   # for upgrade test
+      resource: pxf6-gp6-el8-released
```
Comment (axue-broadcom) [inline concourse/pipelines/templates/jobs/test-multinode-tpl.yml:38]: keep the comment here
```suggestion
      resource: pxf6-gp6-el8-released # for upgrade test
```

File: concourse/scripts/upgrade_pxf.bash (hunk 15 above)
Comment (axue-broadcom) [reply concourse/scripts/upgrade_pxf.bash:57]: Please add a TODO here

File: concourse/pipelines/templates/jobs/test-multinode-tpl.yml [hunk 21]
```diff
@@ -7,7 +7,7 @@
  # x.target     - the target external system type
  # x.distro     - the distro to use for a singlecluster image
  # x.passed     - an array of upstream job names that the artifacts need to pass before this job (optional)
- # x.with_upgrade - 'true' is the job should include an upgrade task from PXF5 to PXF6
+ # x.with_upgrade - 'true' is the job should include an upgrade task from PXF6 to PXF7
```
Comment (denalex) [inline concourse/pipelines/templates/jobs/test-multinode-tpl.yml:10]:
```suggestion
 # x.with_upgrade - 'true' if the job should include an upgrade task from PXF6 to PXF7
```

File: concourse/pipelines/templates/resources/pxf6-gp6-el8-artifact-tpl.yml [hunk 22]
```diff
@@ -1,12 +1,12 @@
-{# Template for a PXF5 GP6 EL7 build artifact resource definition.
+{# Template for a PXF6 GP6 EL7 build artifact resource definition.
```
Comment (denalex) [inline concourse/pipelines/templates/resources/pxf6-gp6-el8-artifact-tpl.yml:1]:
```suggestion
{# Template for a PXF6 GP6 EL8 build artifact resource definition.
```

**Review by denalex — APPROVED** [review]

**Review by axue-broadcom — APPROVED** [review]
